task_processing/plugins/mesos/resource_helpers.py

```diff
@@ -0,0 +1,69 @@
+from typing import Any
+from typing import Dict
+
+from task_processing.plugins.mesos.task_config import MesosTaskConfig
+
+NUMERIC_RESOURCES = frozenset(['cpus', 'mem', 'disk', 'gpus'])
+OTHER_RESOURCES: dict = {'ports': []}
+
+
+def _make_resource_dict():
+    d = dict([(r, 0) for r in NUMERIC_RESOURCES])
+    d.update(OTHER_RESOURCES)
+    return d
+
+
+def get_offer_resources(offer, role: str) -> Dict[str, Any]:
+    """ Get the resources from a Mesos offer
+
+    :param offer: the payload from a Mesos resourceOffer call
+    :param role: the Mesos role we want to get resources for
+    :returns: a mapping from resource name -> available resources for the offer
+    """
+    offer_resources = _make_resource_dict()
+    for resource in offer.resources:
+        if resource.role != role:
+            continue
+
+        if resource.name in NUMERIC_RESOURCES:
+            offer_resources[resource.name] += resource.scalar.value
+        elif resource.name == 'ports':
+            offer_resources['ports'] = [r for r in resource.ranges.range]
```

Comment:
why've you made the numeric_resources additive, wheras the ports are an assignment? I think, for a single offer, you'd expect exactly one resource for each of [``cpu``, ``mem``, ``disk``, ``gpu``] etc, so I don't think you need to do an additive thing? you can just set the value ``offer_resources[resource.name] += resource.scalar.value`` irrespective of whether it is a port or a numeric resource.

Reply:
discussed on slack, fixed